src/main/scala/akka/persistence/jdbc/journal/dao/ByteArrayJournalSerializer.scala

```diff
@@ -17,28 +17,69 @@
 package akka.persistence.jdbc
 package journal.dao
 
+import akka.actor.ActorRef
 import akka.persistence.PersistentRepr
 import akka.persistence.jdbc.serialization.FlowPersistentReprSerializer
-import akka.serialization.Serialization
+import akka.serialization.{ Serialization, SerializerWithStringManifest }
 
 import scala.collection.immutable._
-import scala.util.Try
+import scala.util.{ Failure, Success, Try }
 
-class ByteArrayJournalSerializer(serialization: Serialization, separator: String) extends FlowPersistentReprSerializer[JournalRow] {
+class ByteArrayJournalSerializer(serialization: Serialization, separator: String, writeMessageColumn: Boolean) extends FlowPersistentReprSerializer[JournalRow] {
   override def serialize(persistentRepr: PersistentRepr, tags: Set[String]): Try[JournalRow] = {
-    serialization
-      .serialize(persistentRepr)
-      .map(JournalRow(
+
+    val tryMessageColumn = if (writeMessageColumn) {
+      serialization.serialize(persistentRepr).map(Some.apply)
+    } else {
+      Success(None)
+    }
+
+    tryMessageColumn.map { maybeMessage =>
+      val payload = persistentRepr.payload.asInstanceOf[AnyRef]
+      val serializer = serialization.findSerializerFor(payload)
+      val event = Some(serializer.toBinary(payload))
+      val serManifest = serializer match {
+        case stringManifest: SerializerWithStringManifest =>
+          stringManifest.manifest(payload)
+        case _ if serializer.includeManifest =>
+          payload.getClass.getName
+        case _ => ""
+      }
+      JournalRow(
         Long.MinValue,
         persistentRepr.deleted,
         persistentRepr.persistenceId,
         persistentRepr.sequenceNr,
-        _,
-        encodeTags(tags, separator)))
+        maybeMessage,
+        encodeTags(tags, separator),
+        event,
+        Some(persistentRepr.manifest),
+        Some(serializer.identifier),
+        Some(serManifest),
+        Some(persistentRepr.writerUuid))
+    }
   }
 
   override def deserialize(journalRow: JournalRow): Try[(PersistentRepr, Set[String], Long)] = {
-    serialization.deserialize(journalRow.message, classOf[PersistentRepr])
-      .map((_, decodeTags(journalRow.tags, separator), journalRow.ordering))
+    if (journalRow.event.isDefined && journalRow.serId.isDefined) {
```

Comment:
Case matching on a large case class (JournalRow has 11 fields) would be far less readable than this, and in this case we have 3 options that we want to match on, which would require allocating a 3-tuple just to do a simple if, and I think would reduce the readability too and also be more error prone because two of those options have the same type.